src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,205 @@
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case has {@code null} label, false otherwise
+     */
+    private static boolean hasLiteralNull(DetailAST caseLabel) {
+        return Optional.ofNullable(caseLabel.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.LITERAL_NULL))
+                .isPresent();
+    }
+
+    /**
+     * Checks if the given case AST node has a pattern variable declaration label.
+     * or record pattern definition label.
+     * In this case, if the switch selector evaluates to null,
+     * a {@code NullPointerException} will be thrown.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if case has a pattern in its label
+     */
+    private static boolean hasPatternCaseLabel(DetailAST caseLabel) {
+        return caseLabel.findFirstToken(TokenTypes.RECORD_PATTERN_DEF) != null
+               || caseLabel.findFirstToken(TokenTypes.PATTERN_VARIABLE_DEF) != null
+               || caseLabel.findFirstToken(TokenTypes.PATTERN_DEF) != null;
+    }
+
+    /**
+     * Checks if the given case contains a string in its label.
+     * In this case, if the switch selector evaluates to null,
+     * a {@code NullPointerException} will be thrown.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if switch block contains a string case label
+     */
+    private static boolean hasStringCaseLabel(DetailAST caseLabel) {
+        return hasStringLiteral(caseLabel) || hasStringConcatenatedExpression(caseLabel);
+    }
+
+    /**
+     * Checks if the case label has a string literal.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case label has a string literal
+     */
+    private static boolean hasStringLiteral(DetailAST caseLabel) {
+        return Optional.ofNullable(caseLabel.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.STRING_LITERAL))
+                .isPresent();
+    }
+
+    /**
+     * Checks if the case label has a string in a concatenated expression.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case label has a string in a concatenated expression
+     */
+    private static boolean hasStringConcatenatedExpression(DetailAST caseLabel) {
+        DetailAST ast = caseLabel.getFirstChild().findFirstToken(TokenTypes.PLUS);
+        boolean hasStringInConcatenatedExpression = false;
+        while (ast != null) {
+            if (ast.findFirstToken(TokenTypes.STRING_LITERAL) != null) {
+                hasStringInConcatenatedExpression = true;
+                break;
+            }
+            if (ast.getFirstChild() != null
+                    && ast.getFirstChild().getType() == TokenTypes.PLUS) {
+                ast = ast.getFirstChild();
```

Comment:
Doesn't this seem like a concern that we are going deeper into the tree, without ever returning?

For example: a + b + "c"
````
        |      |  |  |--LITERAL_CASE -> case [4:2]
        |      |  |  |  |--EXPR -> EXPR [4:10]
        |      |  |  |  |  `--PLUS -> + [4:10]
        |      |  |  |  |      |--PLUS -> + [4:8]
        |      |  |  |  |      |  |--IDENT -> a [4:7]
        |      |  |  |  |      |  `--IDENT -> b [4:9]
        |      |  |  |  |      `--STRING_LITERAL -> "c" [4:11]
````

You will miss out on the string literal, because you kept going deep into the plus which just has 2 idents.

Reply:
Cases have to be constants, so a variable shouldn't be possible. However `case 1 + 2 + "Hello" + "lol":` was acceptable for me.

Reply:
> You will miss out on the string literal, because you kept going deep into the plus which just has 2 idents.

before going deeper I check the current plus children if there is a string we will break, if not then go deeper to the next plus. With: 

```Java
 if (ast.findFirstToken(TokenTypes.STRING_LITERAL) != null) {
                hasStringInConcatenatedExpression = true;
                break;
   }
    ...
```

> For example: a + b + "c"


I already have this example and it works correctly

```
        |      |  |  |--LITERAL_CASE -> case [4:2]
        |      |  |  |  |--EXPR -> EXPR [4:10]
        |      |  |  |  |  `--PLUS -> + [4:10]   <---- I will search for string literal before going deep
        |      |  |  |  |      |--PLUS -> + [4:8]
        |      |  |  |  |      |  |--IDENT -> a [4:7]
        |      |  |  |  |      |  `--IDENT -> b [4:9]
        |      |  |  |  |      `--STRING_LITERAL -> "c" [4:11] <--- we found it so we break the loop and return true

```
https://github.com/mahfouz72/checkstyle/blob/76bfa57748922ebb410db76347405a8978856ad5/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingnullcaseinswitch/InputMissingNullCaseInSwitchWithStringLiterals.java#L97-L101

Reply:
> before going deeper I check the current plus children if there is a string we will break

So then this will "break" if I hid string literal better?
`case 1 + 2 + (3 + "lol" + 4) + 5:`

Reply:
we have something similar.

https://github.com/checkstyle/checkstyle/blob/5d14d6b97361177a5df7763b2e6da80118871816/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingnullcaseinswitch/InputMissingNullCaseInSwitchWithStringLiterals.java#L103-L107

but we might have false negatives when the expression becomes very complex. which contains a lot of nested parentheses with a depth of more than 1 and some integers 
something like `(1 + 2 + (1 + (2 + "a" + 3)) + 5)`  
Since we don't do a full search we might miss some string literals if it is hidden well inside a bunch of parenthes.
To have no false negatives we should do a BFS (deep scan like `branchContains()` ) and visit each token starting from the first `plus`,  like we have in our tree walker, Or is it ok to have some false negatives in this weird and uncommon syntax? 

Note that the string will be caught if it is in an expression with any number of  `+`. The false negative may happen in case of nested parens only.

Reply:
> is it ok to have some false negatives in this weird and uncommon syntax

IMO, if we have a false and know it is false, we should fix it. I see this more as an issue with implementation than cases being too complex. If we don't fix this here, this needs to be a new issue. Lines in our inputs would have to be associated to the issue with a "until ...".

> visit each token starting from the first plus, like we have in our tree walker

This is basically what you would need to do if you are not going to explicitly handle the tokens. We have other checks which basically implement their own "treewalker" processing. You can use `TreeWalker`'s code as a basis, but pitest will most likely require it to be trimmed back in certain places.

Reply:
done I used treewalker implementation we now sure that the string will always be detected even if it is hidden in complex expression.